internal/oidc/oauth2.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/ory/fosite/storage"
+	"github.com/ory/fosite/token/jwt"
+
+	"github.com/authelia/authelia/internal/configuration/schema"
+	"github.com/authelia/authelia/internal/middlewares"
+	"github.com/authelia/authelia/internal/utils"
+)
+
+// NewStore creates a new OIDC store.
+func NewStore(config *schema.OpenIDConnectConfiguration) *storage.MemoryStore {
+	clients := make(map[string]fosite.Client)
+
+	for _, v := range config.Clients {
+		clients[v.ID] = &fosite.DefaultClient{
+			ID:            v.ID,
+			Secret:        []byte(v.Secret),
+			RedirectURIs:  v.RedirectURIs,
+			ResponseTypes: v.ResponseTypes,
+			GrantTypes:    v.GrantTypes,
+			Scopes:        v.Scopes,
+		}
+	}
+
+	return &storage.MemoryStore{
+		IDSessions:             make(map[string]fosite.Requester),
+		Clients:                clients,
+		Users:                  map[string]storage.MemoryUserRelation{},
+		AuthorizeCodes:         map[string]storage.StoreAuthorizeCode{},
+		AccessTokens:           map[string]fosite.Requester{},
+		RefreshTokens:          map[string]storage.StoreRefreshToken{},
+		PKCES:                  map[string]fosite.Requester{},
+		AccessTokenRequestIDs:  map[string]string{},
+		RefreshTokenRequestIDs: map[string]string{},
+	}
+}
+
+// InitializeOIDC configures the fasthttp router to provide OIDC.
+func InitializeOIDC(configuration *schema.OpenIDConnectConfiguration, router *router.Router, autheliaMiddleware middlewares.RequestHandlerBridge) {
+	if configuration == nil {
+		return
+	}
+
+	// This is an exemplary storage instance. We will add a client and a user to it so we can use these later on.
+	var store = NewStore(configuration)
+
+	var oidcConfig = new(compose.Config)
+
+	privateKey, err := utils.ParseRsaPrivateKeyFromPemStr(configuration.IssuerPrivateKey)
+	if err != nil {
+		panic(err)
+	}
+
+	oauth2 := compose.ComposeAllEnabled(oidcConfig, store, []byte(configuration.HMACSecret), privateKey)
+
+	// TODO: Add paths for UserInfo, Flush, Logout.
+
+	// TODO: Add OPTIONS handler.
+	router.GET(wellKnownPath, autheliaMiddleware(WellKnownConfigurationHandler))
+
+	router.GET(consentPath, autheliaMiddleware(ConsentGet))
+
+	router.POST(consentPath, autheliaMiddleware(ConsentPost))
+
+	router.GET(jwksPath, autheliaMiddleware(JWKsGet(&privateKey.PublicKey)))
+
+	router.GET(authorizePath, autheliaMiddleware(middlewares.NewHTTPToAutheliaHandlerAdaptor(AuthorizeEndpointGet(oauth2))))
+
+	// TODO: Add OPTIONS handler.
+	router.POST(tokenPath, autheliaMiddleware(middlewares.NewHTTPToAutheliaHandlerAdaptor(tokenEndpoint(oauth2))))
+
+	router.POST(introspectPath, autheliaMiddleware(middlewares.NewHTTPToAutheliaHandlerAdaptor(introspectEndpoint(oauth2))))
+
+	// TODO: Add OPTIONS handler.
+	router.POST(revokePath, autheliaMiddleware(middlewares.NewHTTPToAutheliaHandlerAdaptor(revokeEndpoint(oauth2))))
+}
+
+// A session is passed from the `/auth` to the `/token` endpoint. You probably want to store data like: "Who made the request",
```

Comment:
I think those comments are from a snippet I probably copy pasted. We need to get rid of it.

Reply:
I cleaned up a lot of the comments.